### ISSUE TITLE
`isolated-functions`: selectors should match any function children

### ISSUE THREAD

**Issue by mmkal** [sindresorhus/eslint-plugin-unicorn#2960]
In `isolated-functions`, it's somewhat intuitive to think you can write a selector to match *parents* of functions that should be isolated, e.g.

```json
    "unicorn/isolated-functions": [
      "error",
      {
        "selectors": [
          "CallExpression[callee.property.name=/CodemodeScript/]"
        ]
      }
    ],
```

When you want to match something like

```ts
await client.executeCodemodeScript(async function () {
  await foo()
})
```

But right now it doesn't work, because the selector strictly matches the CallExpression `client.executeCodemodeScript` and not its children (the function passed to it as args).

Today, the fix is to change it to `"CallExpression[callee.property.name=/CodemodeScript/] FunctionExpression"` but then you *also* need to add `ArrowFunctionExpression`, which is easy to forget and then the rule will miss violations if someone arbitrarily uses an arrow function.

Defining a selector on anything but a FunctionExpression or ArrowFunctionExpression will never work, so this change is non-breaking, it just enables a more concise and more intuitive usage.

Implementation would I *think* be fairly easy, we already iterate through the selectors, whenever it matches a node which is not of type FunctionExpression or ArrowFunctionExpression, we can search for children which do have that type. Need to think about functions-within-functions but I think it's doable.

I wrote this rule originally so happy to open a PR.

**Comment by fisker** [comment]
> but then you also need to add ArrowFunctionExpression

You can use `:function`

**Comment by fisker** [comment]
to omit the function selector, you can also write a selector to check parent `[parent.callee.property.name=...]`

**Comment by mmkal** [comment]
Good point about `:function`, I forgot that existed. That definitely makes it less bad
